app/controllers/register/template.hbs

```diff
@@ -0,0 +1,30 @@
+<div class="materialize-me">
+  <div class="container register">
+    <div class="center-align register-logo"></div>
+    <div class="row">
+      <form class="col push-4 s12">
+        <div class="row">
+          <div class="input-field col s12 center-align">
+            <Input @value={{this.passwordText}} type={{if this.showPassword "text" "password"}} @input={{action "valChange"}} class="validate"
+              id="password" />
+            <label for="password">Password</label>
+            <label onclick={{action "showPasswordClick"}} class="right label-icon show-password" for="password"><i class="fas {{if this.showPassword "fa-eye-slash" "fa-eye"}}"></i></label>
+            <StrengthMeter @passwordStrength={{this.passwordStrength}} />
+          </div>
+          <div class="col s12 right-align">
+            <button disabled={{this.disabledButton}} class="waves-effect waves-light btn">Submit</button>
+          </div>
+          <div class="input-field col s12 center-align">
+            <span class="helper-text left-align" data-error="wrong" data-success="right">Strong passwords must contain:</span>
+            <ul class="note-password">
+              <li class="helper-text left-align">Uppercase letter (A-Z)</li>
+              <li class="helper-text left-align">Lowercase letter (a-z)</li>
+              <li class="helper-text left-align">Digit (0-9)</li>
+              <li class="helper-text left-align">Special character (~`!@#$%^&*()+=_-{}[]\|:;”’?/<>,.)</li>
```

Comment:
Are these actually the logic behind zxcvbn's strength calculation?

Reply:
Regarding to my investigation [here](https://raw.githubusercontent.com/dropbox/zxcvbn/master/dist/zxcvbn.js), the logic from zxcvbn is more complicated, zxcvbn built a dictionary for frequently chosen password, and the adjacency_graphs to find the relationship between the characters in password user inputted.  If the password is in the frequency_lists, or if zxcvbn found that your password have a rule , your password will have a low score. If you following above rules the password is easier to achieve the high score.